tests/attacks/test_pixel_attack.py

```diff
@@ -0,0 +1,174 @@
+# MIT License
+#
+# Copyright (C) IBM Corporation 2018
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the  rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module tests the Pixel Attack.
+The Pixel Attack is a generalisation of One Pixel Attack.
+
+| One Pixel Attack Paper link:
+    https://ieeexplore.ieee.org/abstract/document/8601309/citations#citations
+    (arXiv link: https://arxiv.org/pdf/1710.08864.pdf)
+| Pixel Attack Paper link:
+    https://arxiv.org/abs/1906.06026
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import unittest
+
+import numpy as np
+
+from tests.utils import TestBase
+from tests.utils import get_classifier_tf, get_classifier_kr, get_classifier_pt
+
+from art.attacks import PixelAttack
+from art.utils import get_labels_np_array
+
+logger = logging.getLogger(__name__)
+
+
+class TestPixelAttack(TestBase):
+    """
+    TODO: Write Comment
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        TODO: Write Comment
+        """
+        super().setUpClass()
+
+        cls.n_test = 2
+        cls.x_test_mnist = cls.x_test_mnist[0:cls.n_test]
+        cls.y_test_mnist = cls.y_test_mnist[0:cls.n_test]
+
+    def test_keras_mnist(self):
+        """
+        TODO: Write Comment
+        """
+        classifier = get_classifier_kr()
+        self._test_attack(
+            classifier,
+            self.x_test_mnist,
+            self.y_test_mnist,
+            False)
+
+    def test_tensorflow_mnist(self):
+        """
+        TODO: Write Comment
+        """
+        classifier, sess = get_classifier_tf()
+        self._test_attack(
+            classifier,
+            self.x_test_mnist,
+            self.y_test_mnist,
+            False)
+
+    def test_pytorch_mnist(self):
+        """
+        TODO: Write Comment
+        """
+        x_test = np.swapaxes(self.x_test_mnist, 1, 3).astype(np.float32)
```

Comment:
```suggestion
        x_test = np.reshape(self.x_test_mnist, (self.x_test_mnist.shape[0], 28, 28, 1)).astype(np.float32)
```

Reply:
Please note that the `get_classifier_pt` expects an input of shape `(1,28,28)` and not `(28,28,1)`
The suggestion would, therefore, create a contradiction to the expected shape.

https://github.com/IBM/adversarial-robustness-toolbox/blob/e6d0b53a16bca37712a78ad916018c67a5951abe/tests/utils_test.py#L570

Reply:
Sorry, could you please replace it with this line, this time the shape should be correct:
```
x_test = np.reshape(self.x_test_mnist, (self.x_test_mnist.shape[0], 1, 28, 28)).astype(np.float32)
```
We use reshape instead of swapping axes to make sure that the PyTorch testing classifier is predicting the same as the Keras and TensorFlow testing classifiers.

Reply:
No Problem. I have updated the code with the same.